demo/clusterctl-examples/README.md

```diff
@@ -0,0 +1,150 @@
+# Table Of Contents
+
+- [Overview](#overview)
+  - [OS Image](#os_image)
+  - [Network Configurations](#network_configurations)
+  - [Network Access](#network_access)
+  - [User Accounts](#user_accounts)
+- [Prerequsites](#prerequisites)
+- [How-To](#how_to)
+
+# Overview <a name="overview" />
+
+This directory contains sample [clusterctl templates][clusterctl-templates] for
+both [cri-o][cri-o] and [rke2][rke2] clusters. The boostrapping and
+configuration of these clusters will be largely depended on the following:
+
+## OS Image <a name="os_image" />
+
+The type of OS image (i.e. SUSE-based, Ubuntu-based, etc) and what's on that
+image (i.e. packages pre-installed, package availability, networking setup,
+security setup, user accounts setup, etc) will determine what the cluster
+bootstrapping commands will look like.
+
+## Network Configurations <a name="network_configurations" />
+
+The network configuration is very much environment-specific. *The NIC names
+for the baremetal nodes must be consistent as they are sharing the same
+Metal3DataTemplate*. For example, we can't have a NIC in one node named
+`eno1f0` while the other node have it as `eth0`.
+
+Furthermore, the configuration of IP pools, gateways, DNS, and routing table
+requires indepth knowledge of infrastructure networking setup.
```

Comment:
We should document the networking setup assumed in the examples. That will make it easy for people to understand what they need to change to fit their own environment.

Reply:
Done